dali/operators/image/resize/resize_attr_base.h

```diff
@@ -103,8 +103,20 @@ void CalculateSampleParams(ResizeParams &params, SmallVector<float, 3> requested
       // and real output size instead.
       adjustment = clamp(adjustment, -10.0, 10.0);
 
-      // keep center of the ROI - adjust the edges
-      double center = (params.src_lo[d] + params.src_hi[d]) * 0.5;
+      double center;
+      if (alignment.empty() || alignment[d] == 0) {
+        // keep center of the ROI - adjust the edges
+        center = (params.src_lo[d] + params.src_hi[d]) * 0.5;
+      } else if (alignment[d] == -1) {
+        // keep start of the ROI
+        center = params.src_lo[d];
+      } else if (alignment[d] == 1) {
+        // keep end of the ROI
+        center = params.src_hi[d];
+      } else {
+        DALI_FAIL(make_string("Unsupported alignment value ", alignment[d],
+                              ". Supported values are 0, -1, 1"));
+      }
```

Comment:
To consider:
```suggestion
      } else if (alignment[d] < -1) {
        // keep start of the ROI
        center = params.src_lo[d];
      } else if (alignment[d] > 1) {
        // keep end of the ROI
        center = params.src_hi[d];
      }
```
Having a sign (as opposed to exact value) define the behavior may simplify the client code. The usage of sign is not uncommon (see `strcmp` or the "spaceship" operator `<=>`).

Reply:
A random thought: how about having an alignment as a `float` 0..1? 0.5 would denote the center.
```C++
center = (1 - alignment[d]) * params.src_lo[d] + alignment[d] * params.src_hi[d]
```